### ISSUE TITLE
How to understand frame correction operations?

### ISSUE THREAD

**Issue by Lwzzzzzz** [zyn213/TempRMOT#6]
Thank you for your work! I notice that frame correction #2 only occurs during the generation of GT files in inference. And I can't find any similar operations in the training code. So why do we only need to delay one frame backward during inference. Does it mean that only videos used for inference have annotation misalignment?

Or, if any corrections have made during the training process, could you please help me find the specific location?

**Comment by zyn213** [comment]
Hi, thanks for your interest.

> Does it mean that only videos used for inference have annotation misalignment?


And the answer is yes. TransRMOT does not experience frame misalignment issues during the training phase. This issue only arises when obtaining GT during inference.
You only need to simply modify the gt.txt file and do eval with the official predict.txt file from [TransRMOT](https://github.com/wudongming97/RMOT) to get a result of 46.56.

**Comment by Lwzzzzzz** [comment]
Thanks! I understand what you mean. But I would like to confirm again that when using the Refer-kitti/2 dataset for training, I don't need to delay by one frame. Instead, when using videos 5, 11, 13, and 19 for testing, I need to delay GT by one frame, right?

In other words, I am concerned if you have also inserted delayed frame operations similar to inference in the training code of TransRMOT or TempRMOT?

**Comment by zyn213** [comment]
No, our operation is not delaying GT by one frame, we are aligning frames. 
I will use the source code in TransRMOT as an example:
Assume that the current frame is frame 0(**id**). When writing the GT file, it marks frame 0 as frame 1 (id+1)
![image](https://github.com/zyn213/TempRMOT/assets/94900013/44d2fee0-66e4-4c5f-88ae-5a697ee667f7)
When getting GT, it **gets the GT of the first frame (id+1)** and marks it as the first frame (id+1)
![image](https://github.com/zyn213/TempRMOT/assets/94900013/426af156-79ed-40e6-a9da-d862f5ce4f6c)

The correct approach should be to first obtain the GT of frame 0, and then mark it as 1 (id+1)
![image](https://github.com/zyn213/TempRMOT/assets/94900013/82825032-3ab6-4561-a4d5-65d1fdd0cd0a)

**Comment by Lwzzzzzz** [comment]
Thank you for your patience! Perhaps my expression has caused you to misunderstand. However, I have already figured it out on my own. In the **Detector.detect()** of the inference.py, the output result is marked as the next frame, so it is necessary to also mark GT as the next frame.

![image](https://github.com/zyn213/TempRMOT/assets/46078150/be3b0233-fb69-41c3-a32d-f8135d1dae90)

However, this situation does not exist during the training process, so when inputting the frame 0, the output can directly calculate the loss with the GT of frame 0. I have been worried before whether I also need to calculate the loss between the current frame image and the previous frame GT during training.
